### ISSUE TITLE
[bug] `roots` is not defined when using eslint-config

### ISSUE THREAD

**Issue by joshuafredrickson** [roots/bud#2478]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Describe the issue

When running with the following `.eslintrc.cjs` in an otherwise vanilla Sage config, I receive an undefined error on `roots` in `scripts/editor.js`:

```js
module.exports = {
  root: true,
  extends: ['@roots/eslint-config/sage'],
};
```

### Expected Behavior

No error. 

### Actual Behavior

Error: 
> 'roots' is not defined. eslint([no-undef](https://eslint.org/docs/latest/rules/no-undef))

<img width="865" alt="image" src="https://github.com/roots/bud/assets/3533660/b796912c-5a36-4d5f-821c-f6af436e4c87">

### Steps To Reproduce

Run fresh Sage install with `@roots/eslint-config:6.16.1` and the `.eslintrc.cjs` file mentioned above. Error will display in VS Code when viewing `scripts/editor.js`.

### version

6.16.1

**Comment by kellymears** [comment]
Your PR looks good. I think we should also add it to some sort of importable typings. Even something simple like:

```ts
declare const roots: {
  register: {
    blocks: any
    // ...
  }
}
```

Also probably makes sense in `sources/create-bud-app/templates/types.d.hbs`.

But, that can be handled in a separate PR. I'll merge your PR as soon as I get a chance to verify it.
